runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerPlugin.java

```diff
@@ -474,62 +467,68 @@ public void onChatMessage(ChatMessage event)
 		final int regionID = client.getLocalPlayer().getWorldLocation().getRegionID();
 		if (HESPORI_REGION == regionID && message.equals(HESPORI_LOOTED_MESSAGE))
 		{
-			eventType = HESPORI_EVENT;
+			mostRecentEvent = LootEvent.HESPORI;
 			takeInventorySnapshot();
 			return;
 		}
 
 		if (GAUNTLET_LOBBY_REGION == regionID && message.equals(GAUNTLET_LOOTED_MESSAGE))
 		{
-			eventType = GAUNTLET_EVENT;
+			mostRecentEvent = LootEvent.GAUNTLET;
 			takeInventorySnapshot();
 			return;
 		}
 
+		final Matcher pickpocketMatcher = pickpocketingRegex.matcher(message);
+		if (pickpocketMatcher.matches())
+		{
+			// H.A.M. Members are just listed as "men" and "women" in the chat box, so we have to make an exception
+			String pickpocketTarget = pickpocketingHamMembers ? "H.A.M. Member" : WordUtils.capitalize(pickpocketMatcher.group("target"));
+
+			if (LootEvent.PICKPOCKET_EVENT_LOOKUP_TABLE.containsKey(pickpocketTarget))
+			{
+				mostRecentEvent = LootEvent.PICKPOCKET_EVENT_LOOKUP_TABLE.get(pickpocketTarget);
+				takeInventorySnapshot();
+				return;
+			}
+		}
+
 		// Check if message is for a clue scroll reward
 		final Matcher m = CLUE_SCROLL_PATTERN.matcher(Text.removeTags(message));
 		if (m.find())
 		{
 			final String type = m.group(1).toLowerCase();
-			switch (type)
-			{
-				case "beginner":
-					eventType = "Clue Scroll (Beginner)";
-					break;
-				case "easy":
-					eventType = "Clue Scroll (Easy)";
-					break;
-				case "medium":
-					eventType = "Clue Scroll (Medium)";
-					break;
-				case "hard":
-					eventType = "Clue Scroll (Hard)";
-					break;
-				case "elite":
-					eventType = "Clue Scroll (Elite)";
-					break;
-				case "master":
-					eventType = "Clue Scroll (Master)";
-					break;
-			}
+			mostRecentEvent = LootEvent.CLUE_SCROLL_LOOKUP_TABLE.get(type);
 		}
 	}
 
 	@Subscribe
 	public void onItemContainerChanged(ItemContainerChanged event)
 	{
-		if (CHEST_EVENT_TYPES.containsValue(eventType)
-			|| HERBIBOAR_EVENT.equals(eventType)
-			|| HESPORI_EVENT.equals(eventType)
-			|| GAUNTLET_EVENT.equals(eventType))
+		if (mostRecentEvent == null)
+		{
+			return;
+		}
+
+		if (LootEvent.DIRECT_TO_INVENTORY_EVENTS.contains(mostRecentEvent))
 		{
 			if (event.getItemContainer() != client.getItemContainer(InventoryID.INVENTORY))
 			{
 				return;
 			}
 
-			processChestLoot(eventType, event.getItemContainer());
-			eventType = null;
+			processChestLoot(mostRecentEvent, event.getItemContainer());
+			mostRecentEvent = null;
+		}
+	}
+
+	@Subscribe
+	public void onMenuOptionClicked(MenuOptionClicked event)
+	{
+		// H.A.M. members show up differently in the chat box, so we need to specially track them
+		if (event.getMenuOption().equals("Pickpocket"))
+		{
+			pickpocketingHamMembers = Text.removeTags(event.getMenuTarget()).equals("H.A.M. Member");
```

Comment:
You could probably just store the full name here and always use it to avoid this issue in the future with other things.

Reply:
Store the name of the last pickpocket target? That was my first implementation, but I think it may have some issues on its own:

* there's a very small (though not non-existent) risk of race conditions, since players can select different menu options while mid-pickpocket, so the loot might potentially be recorded under the wrong name.
* some pickpocket-able NPCs have randomized names, though their pickpocketing message is the same. e.g. I believe all the Prifddinas elves have different names, but still display as "you pick the elf's pocket", so there'd be a lot of book-keeping required to map all those names.

I do agree that a specific boolean just for ham members is a bit hacky, so I'll consider more extensible alternatives before pushing the newest changes